routemaster/config/loader.py

```diff
@@ -149,36 +151,70 @@ def _load_feed_config(yaml: Yaml) -> FeedConfig:
     return FeedConfig(name=yaml['name'], url=yaml['url'])
 
 
-def _load_state(path: Path, yaml_state: Yaml) -> State:
+def _load_state(path: Path, yaml_state: Yaml, feed_names: List[str]) -> State:
     if 'action' in yaml_state and 'gate' in yaml_state:  # pragma: no branch
         raise ConfigError(  # pragma: no cover
             f"State at path {'.'.join(path)} cannot be both a gate and an "
             f"action.",
         )
 
     if 'action' in yaml_state:
-        return _load_action(path, yaml_state)
+        return _load_action(path, yaml_state, feed_names)
     elif 'gate' in yaml_state:  # pragma: no branch
-        return _load_gate(path, yaml_state)
+        return _load_gate(path, yaml_state, feed_names)
     else:
         raise ConfigError(  # pragma: no cover
             f"State at path {'.'.join(path)} must be either a gate or an "
             f"action.",
         )
 
 
-def _load_action(path: Path, yaml_state: Yaml) -> Action:
+def _validate_context_lookups(
+    path: Path,
+    lookups: Iterable[str],
+    feed_names: List[str],
+) -> None:
+    # Changing this? Also change context lookups in
+    # `routemaster.context.Context`
+    VALID_TOP_LEVEL = ('feeds', 'history', 'metadata')
+
+    for lookup in lookups:
+        location, *rest = lookup.split('.')
+
+        if location not in VALID_TOP_LEVEL:
+            raise ConfigError(
+                f"Invalid context lookup at {'.'.join(path)}: key {lookup} "
+                f"must start with one of 'feeds', 'history' or 'metadata'.",
```

Comment:
Can we re-use `VALID_TOP_LEVEL` for this message?

Reply:
Yup, 3085173.